duckduckgo/Android
# Reopen Android: Improve lock contention issues (performance)

Task/Issue URL: https://app.asana.com/0/0/1205529944668184/f

### Description
Revert "Revert "Minimise use of default() dispatcher".
See https://app.asana.com/0/1200905986587319/1204763838216666/f

### Steps to test this PR
Smoke tests for every feature

See https://github.com/duckduckgo/Android/pull/3463

### NO UI changes
